clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/AppEngineServerGroupNameResolver.groovy

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.appengine.deploy
+
+import com.google.api.services.appengine.v1.model.Version
+import com.netflix.frigga.Names
+import com.netflix.spinnaker.clouddriver.appengine.security.AppEngineNamedAccountCredentials
+import com.netflix.spinnaker.clouddriver.helpers.AbstractServerGroupNameResolver
+
+import java.text.SimpleDateFormat
+
+class AppEngineServerGroupNameResolver extends AbstractServerGroupNameResolver {
+  private static final String PHASE = "DEPLOY"
+  private static final SIMPLE_DATE_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSSX"
+
+  private final String project
+  private final AppEngineNamedAccountCredentials credentials
+
+  AppEngineServerGroupNameResolver(String project, AppEngineNamedAccountCredentials credentials) {
+    this.project = project
+    this.credentials = credentials
+  }
+
+  @Override
+  String getPhase() {
+    return PHASE
+  }
+
+  @Override
+  String getRegion() {
```

Comment:
Maybe we should think about what this region should be... `"appengine"` is going to look funny in certain places like the UI.